crates/wash-lib/src/context/fs.rs

```diff
@@ -29,42 +31,64 @@ impl Deref for ContextDir {
 }
 
 impl ContextDir {
-    /// Creates a new ContextDir, erroring if it is unable to access or create the given directory.
-    ///
-    /// This should be the directory and not the file (e.g. "/home/foo/.wash/contexts")
-    pub fn new(path: impl AsRef<Path>) -> Result<ContextDir> {
-        let p = path.as_ref();
-        let exists = p.exists();
-        if exists && !p.is_dir() {
-            anyhow::bail!("{} is not a directory (or cannot be accessed)", p.display())
+    /// Creates a new ContextDir at ~/.wash/contexts
+    pub fn new() -> Result<ContextDir> {
+        Self::new_with_dir(None::<&Path>)
+    }
+
+    /// Creates a new ContextDir at the specified path. If a path is not provided, defaults to ~/.wash/contexts
+    pub fn new_with_dir(path: Option<impl AsRef<Path>>) -> Result<ContextDir> {
```

Comment:
This is really similar to the other case of going from a context name or the default but how do you feel about:

NIT: `ContextDir::from_dir` ?

I also like the `TryFrom` approach, but this unfortunately clashes w/ named contexts and file paths since they both can be interpreted as paths
```rust
impl<T> TryFrom<Option<T>> for ContextDir where T: AsRef<std::path::Path> {
    type Error = anyhow::Error;
    fn try_from(other: Option<T>) -> anyhow::Result<ContextDir> { ... }
}
```

Reply:
I like `from_dir` 👍